base/abstractarray.jl

```diff
@@ -422,6 +422,21 @@ function _maxlength(A, B, C...)
     max(length(A), _maxlength(B, C...))
 end
 
+# eachindex(A, d) and eachindex(A, (:, j))
+eachindex(A::AbstractArray, d::Integer) = 1:size(A, d)
```

Comment:
Hopefully keyword arguments is another one of those fixable performance problems that will go away at some point like slow anonymous functions recently did. `dims` as a positional argument changes behavior so dramatically, it really feels like a bad Matlab-ism that we should try to avoid.